packages/@aws-cdk/aws-servicecatalog/test/product.test.ts

```diff
@@ -253,9 +253,60 @@ describe('Product', () => {
     expect(snapshotExists).toBe(true);
   }),
 
-  test('fails product test from product stack when template changes and locked', () => {
+  test('all product stack versions from file system', () => {
+    const productStack = new servicecatalog.ProductStack(stack, 'ProductStack');
+
+    const productStackHistory = new ProductStackHistory(stack, 'MyProductStackHistory', {
+      directory: 'product-stack-snapshots/all-versions-test',
+      productStack: productStack,
+      currentVersionName: 'v1',
+      currentVersionLocked: false,
+    });
+
+    new sns.Topic(productStack, 'SNSTopicProductStack');
+
+    new servicecatalog.CloudFormationProduct(stack, 'MyProduct', {
+      productName: 'testProduct',
+      owner: 'testOwner',
+      productVersions: [
+        productStackHistory.currentVersion(),
+      ],
+    });
+
+    app.synth();
+    const versions = productStackHistory.allVersions();
+    expect(versions.length).toBe(1);
+    expect(versions[0].productVersionName).toEqual('v1');
```

Comment:
This is a good start. Since this is supposed to be getting allVersions, can you please extend the test to load more than just the current version. Keeping in mind that the current version has a special code path in `allVersions()`.